src/api.v2/controllers/subsetController.js

```diff
@@ -10,66 +9,86 @@ const UserAccess = require('../model/UserAccess');
 
 const logger = getLogger('[SubsetController] - ');
 
-const runSubset = async (req, res) => {
-  const { experimentId } = req.params;
-
-  logger.log(`Starting subset for experiment ${experimentId}`);
-
+const runSubset = async (stateMachineParams, authorization) => {
   const {
-    params: { experimentId: parentExperimentId },
-    body: { name, cellSetKeys },
-    user: { sub: userId },
-  } = req;
+    experimentId: parentExperimentId,
+    name,
+    userId,
+  } = stateMachineParams;
 
-  logger.log(`Creating experiment to subset ${parentExperimentId}`);
+  // Doesn't exist initially
+  let { subsetExperimentId } = stateMachineParams;
 
-  let subsetExperimentId;
-  await sqlClient.get().transaction(async (trx) => {
-    subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
-    await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
+  // Upon retrying the subset experiment id already exists,
+  // and we don't want to create a new experiment
+  if (!subsetExperimentId) {
+    logger.log(`Creating experiment to subset ${parentExperimentId}`);
 
-    await new ExperimentParent(trx).create(
-      { experiment_id: subsetExperimentId, parent_experiment_id: parentExperimentId },
-    );
-  });
+    await sqlClient.get().transaction(async (trx) => {
+      subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
+      await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
 
-  // Samples are not created here, we add them in handleResponse of SubsetSeurat
+      await new ExperimentParent(trx).create(
+        { experiment_id: subsetExperimentId, parent_experiment_id: parentExperimentId },
+      );
+    });
 
-  logger.log(`Created ${subsetExperimentId}, subsetting experiment ${parentExperimentId} to it`);
+    // Samples are not created here, we add them in handleResponse of SubsetSeurat
+    logger.log(`Created ${subsetExperimentId}, subsetting experiment ${parentExperimentId} to it`);
+  }
 
   const {
     processingConfig: parentProcessingConfig,
   } = await new Experiment().findById(parentExperimentId).first();
 
   const { stateMachineArn, executionArn } = await createSubsetPipeline(
-    parentExperimentId,
+    {
+      ...stateMachineParams,
+      subsetExperimentId,
+    },
     subsetExperimentId,
-    name,
-    cellSetKeys,
     parentProcessingConfig,
-    req.headers.authorization,
+    authorization,
   );
 
   const newExecution = {
-    state_machine_arn: stateMachineArn,
+    stateMachine_arn: stateMachineArn,
```

Comment:
This should be failing I think, `stateMachine_arn` is not the name of the sql column, it's `state_machine_arn`